myTrip/comment/models.py

```diff
@@ -22,56 +38,134 @@ def get_by_id(comment_id):
         Args:
             comment_id (int): comment id.
         Returns:
-            QuerySet<Comment>: QuerySet of Comment.
+            Object<Comment>: Object of Comment.
         """
         try:
             return Comment.objects.get(id=comment_id)
         except ObjectDoesNotExist:
             return None
 
+    @staticmethod
+    def get_by_trip_id(trip_id):
+        """
+        Get Comments with given trip id
+        Args:
+            trip_id (int): user id foreign key to Trip.
+        Returns:
+            QuerySet<Comment>: QuerySet of Comments.
+        """
+        try:
+            return Comment.objects.filter(trip=trip_id)
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def get_by_checkpoint_id(checkpoint_id):
+        """
+        Get Comments with given trip id
+        Args:
+            checkpoint_id (int): user id foreign key to Trip.
+        Returns:
+            QuerySet<Comment>: QuerySet of Comments.
+        """
+        try:
+            return Comment.objects.filter(checkpoint=checkpoint_id)
+        except ObjectDoesNotExist:
```

Comment:
If there are no elements 'filter' will return empty QuerySet, but it never raises 'ObjectDoesNotExist'